hypothesis-python/src/hypothesis/internal/conjecture/datatree.py

```diff
@@ -67,6 +67,83 @@ class Conclusion:
     interesting_origin = attr.ib()
 
 
+# The number of max children where, beyond this, it is practically impossible
+# for hypothesis to saturate / explore all children nodes in a reasonable time
+# frame. We use this to bail out of expensive max children computations early,
+# where the numbers involved are so large that we know they will be larger than
+# this number.
+#
+# Note that it's ok for us to underestimate the number of max children of a node
+# by using this. We just may think the node is exhausted when in fact it has more
+# possible children to be explored. This has the potential to finish generation
+# early due to exhausting the entire tree, but that is quite unlikely: (1) the
+# number of examples would have to be quite high, and (2) the tree would have to
+# contain only one or two nodes, or generate_novel_prefix would simply switch to
+# exploring another non-exhausted node.
+#
+# Also note that we may sometimes compute max children above this value. In other
+# words, this is *not* a hard maximum on the computed max children. It's the point
+# where further computation is not beneficial - but sometimes doing that computation
+# unconditionally is cheaper than estimating against this value.
+#
+# The one case where this may be detrimental is fuzzing, where the throughput of
+# examples is so high that it really may saturate important nodes. We'll cross
+# that bridge when we come to it.
+MAX_CHILDREN_EFFECTIVELY_INFINITE = 100_000
+
+
+def compute_max_children(kwargs, ir_type):
+    if ir_type == "integer":
+        min_value = kwargs["min_value"]
+        max_value = kwargs["max_value"]
+        if min_value is not None and max_value is not None:
+            return max_value - min_value + 1
+        if min_value is not None:
+            return (2**127) - min_value
+        if max_value is not None:
+            return (2**127) + max_value
+        return 2**128 - 1
```

Comment:
Two of these returns are off by one? Either the first and third, or the second and last.

Dropping the conditional returns, I'd have thought this is correct:
```
if min_value is None:
    min_value = -(2**127)
if max_value is None:
    max_value = 2**127 - 1
return max_value - min_value + 1
```
but it differs by one if `min_value is None` (second and last return).


(drive-by comment without understanding the context, feel free to ignore)

Reply:
I *think* this code is correct. `min_value = -(2**127) + 1`, not `-(2**127)`, may be the source of the off by one error here. For sanity: https://gist.github.com/tybug/818ac74ae485fc24d97437bd69ac4d0d

Your version is more readable though, so I've switched to it (with a changed `min_value`).

Drive by comments much appreciated — thanks!